synapse/models/infotech.py

```diff
@@ -303,6 +450,46 @@ def getBaseModels():
                 )),
+                ('it:prod:soft', {}, (
+                    ('name', {'ptype': 'str:lwr', 'ro': 1, 'req': 1,
+                              'doc': 'Name of the software'}),
+                    ('desc', {'ptype': 'str:txt', 'doc': 'A description of the software'}),
+                    ('desc:short', {'ptype': 'str:lwr', 'doc': 'A short description of the software'}),
+                    ('author:org', {'ptype': 'ou:org', 'doc': 'Organization responsible for the software', }),
+                    ('author:acct', {'ptype': 'inet:web:acct', 'doc': 'Web user responsible for the software', }),
+                    ('author:person', {'ptype': 'ps:person', 'doc': 'Person responsible for the software', }),
+                    ('url', {'ptype': 'inet:url', 'doc': 'URL relevant for the software', }),
+                )),
+
+                ('it:prod:softver', {}, (
+                    ('software', {'ptype': 'it:prod:soft', 'req': 1, 'ro': 1,
+                                  'doc': 'Software associated with this version instance.'}),
+                    ('verstr', {'ptype': 'it:dev:str', 'req': 1, 'ro': 1, 'ex': '1.0.2'
+                                'Version string associated with this version instance.'}),
+                    ('verstr:norm', {'ptype': 'str:lwr', 'doc': 'Normalized version of the version string.'}),
+                    ('arch', {'ptype': 'it:dev:str', 'doc': 'Software architecture.'}),
+                    ('semver', {'ptype': 'it:semver', 'doc': 'System normalized semantic version number.', }),
+                    ('semver:major', {'ptype': 'int', 'doc': 'Version major number', }),
+                    ('semver:minor', {'ptype': 'int', 'doc': 'Version minor number', }),
+                    ('semver:patch', {'ptype': 'int', 'doc': 'Version patch number', }),
+                    ('semver:prerelease', {'ptype': 'str:txt', 'doc': 'Semver prerelease string.', }),
+                    ('semver:build', {'ptype': 'str:txt', 'doc': 'Semver build string.', }),
+                    ('url', {'ptype': 'inet:url',
+                             'doc': 'URL where a specific version of the software is available from'}),
+                )),
+
+                ('it:hostsoft', {}, (
+                    ('host', {'ptype': 'it:host', 'ro': 1, 'req': 1,
+                              'doc': 'Host with the software', }),
+                    ('softver', {'ptype': 'it:prod:softver', 'ro': 1, 'req': 1,
+                                  'doc': 'Software on the host', }),
+                    ('seen:min', {'ptype': 'time:min',
+                                  'doc': 'Minimum time the software was seen on the host', }),
+                    ('seen:max', {'ptype': 'time:max',
+                                  'doc': 'Maximum time the software was seen on the host', }),
+                    ('path', {'ptype': 'file:path',
```

Comment:
not sure what this one means exactly...  most software will be many files.  does this mean an installation directory?

Reply:
the idea was that we could represent "software was installed at a host at ``/blah/path``"; where that could represent where a software package was installed (without requireing a specific exe).  For example, something like ``C:\Python27`` or ``/usr/bin/python`` could be a path to a software package installed on a host.